src/liballoc/vec.rs

```diff
@@ -992,7 +992,7 @@ impl<T> Vec<T> {
     #[stable(feature = "rust1", since = "1.0.0")]
     pub fn insert(&mut self, index: usize, element: T) {
         let len = self.len();
-        assert!(index <= len);
+        assert!(index <= len, "index out of bounds: the len is {} but the index is {}", len, index);
```

Comment:
Given that the bounds check for `insert` functions is different from indexing (`<=` rather than `<`), I wonder if it makes sense to also make the error different?

Reply:
Also, if we want to be *really* sure this behaves like indexing, we could make `core::panicking::panic_bounds_check` public and call it (though until the next bootstrap bump, that requires conditional compilation unfortunately).

I am not sure if that's worth it, though.

Reply:
> Given that the bounds check for insert functions is different from indexing (<= rather than <), I wonder if it makes sense to also make the error different?

I was thinking on how to check all three boxes: 
- don't fall into bloated error message
- introduce argument values
- keep it consistent with everything else (basically `[]` call)

Other than that, I think bringing both conditions and parameters might work better. However, if we want to add some human-readable explanation, it gets really big, like that 
```rust
assert!("index out of bounds, index should be <= len, got {} <= {}", index, len);
```

Reply:
Well, now I'm thinking that is has an appropriate size, but then it will require `core::panicking::panic_bounds_check` to be updated, to bring condition there too

Reply:
No the condition would have to stay here.

Reply:
Can you elaborate? I think I didn't get the context

Reply:
I'm just saying that function must not change.

But in the end it probably is better to just do what you did, and not try to literally share the message with indexing... I was just bringing up the possibility.